sources/procfs.go

```diff
@@ -376,60 +347,31 @@ func getJobStatsByOperation(jobBlock string, jobID string, operation string) (me
 }
 
 func getJobStats(jobBlock string, jobID string) (metricList []lustreJobsMetric, err error) {
-	readStatsList, err := getJobStatsByOperation(jobBlock, jobID, "read")
-	if err != nil {
-		return nil, err
-	}
-	if readStatsList != nil {
-		for _, item := range readStatsList {
-			metricList = append(metricList, item)
+	operations := []string{"read", "write"}
+	for _, operation := range operations {
+		statsList, err := getJobStatsByOperation(jobBlock, jobID, operation)
+		if err != nil {
+			return nil, err
 		}
-	}
-
-	writeStatsList, err := getJobStatsByOperation(jobBlock, jobID, "write")
-	if err != nil {
-		return nil, err
-	}
-	if writeStatsList != nil {
-		for _, item := range writeStatsList {
-			metricList = append(metricList, item)
+		if statsList != nil {
+			for _, item := range statsList {
+				metricList = append(metricList, item)
+			}
 		}
 	}
 
 	return metricList, nil
 }
 
 func getJobNum(jobBlock string) (jobID string, err error) {
-	numbersRegex, err := regexp.Compile("[0-9]*.[0-9]+|[0-9]+")
-	if err != nil {
-		return "", err
-	}
-	jobIDRegex, err := regexp.Compile("job_id: .*")
-	if err != nil {
-		return "", err
-	}
-
-	jobID = jobIDRegex.FindString(jobBlock)
-	if len(jobID) == 0 {
-		return "", nil
-	}
-	jobID = numbersRegex.FindString(jobID)
-	if len(jobID) == 0 {
-		return "", nil
-	}
-
+	jobID = regexCaptureString("job_id: .*", jobBlock)
+	jobID = regexCaptureString("[0-9]*.[0-9]+|[0-9]+", jobID)
 	return strings.Trim(jobID, " "), nil
 }
 
 func parseJobStatsText(jobStats string) (metricList []lustreJobsMetric, err error) {
-	pattern := "(?ms:job_id:.*?(-|\\z))"
-	jobRegex, err := regexp.Compile(pattern)
-	if err != nil {
-		return nil, err
-	}
-
-	jobs := jobRegex.FindAllString(jobStats, -1)
-	if len(jobs) == 0 {
+	jobs := regexCaptureStrings("(?ms:job_id:.*?(-|\\z))", jobStats)
+	if len(jobs) < 0 {
```

Comment:
You're no longer returning nil, nil in cases where len(jobs) is equal to zero. Is that intentional?